synapse/rest/client/v1/room.py

```diff
@@ -874,16 +874,19 @@ async def on_PUT(self, request, room_id, user_id):
         # Limit timeout to stop people from setting silly typing timeouts.
         timeout = min(content.get("timeout", 30000), 120000)
 
+        # Defer getting the typing handler since it will raise on workers.
+        typing_handler = self.hs.get_typing_writer_handler()
```

Comment:
This is annoying, but the other option I came up with was to *not* register the endpoint on other works which people said was more confusing.